### PR TITLE
Com 209 accordion component

### PR DESCRIPTION
**Pull request by dawinValenzuela**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

**Storybook Link** (Link to a local storybook url to review new functionality)

### CODE REVIEW

**Comment by dawinValenzuela** [comment]
@here just a cooment, Chakra doesn't export and AccordionIconProps, so I decide to use IconProps for the AccordionIcon component but I'm having this issue, seems like a coalition with the props. maybe someone can help me with this. 

or maybe we can just export dirrectly the AccordionIcon like this.

Thank you.

`Accordion.Icon = AccordionIcon`

image

**Comment by TheEnmanuel23** [comment]
> @here just a cooment, Chakra doesn't export and AccordionIconProps, so I decide to use IconProps for the AccordionIcon component but I'm having this issue, seems like a coalition with the props. maybe someone can help me with this.
> 
> or maybe we can just export dirrectly the AccordionIcon like this.
> 
> Thank you.
> 
> `Accordion.Icon = AccordionIcon`
> 
> image

IMHO, I think that in order to avoid getting that error, you might not create the new variable `AccordionIcon`  but you might pass it to the Accordion.Icon declaration and you will get basically the same.
```jsx
Accordion.Icon = ChakraAccordionIcon;
```

**Comment by dawinValenzuela** [comment]
I'm agree with you @TheEnmanuel23. PR updated with that.

File: packages/components/src/components/primitives/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+const AccordionItem: FC<AccordionItemProps> = (props) => (
+  <ChakraAccordionItem {...props} />
+);
+const AccordionButton: FC<AccordionButtonProps> = (props) => (
+  <ChakraAccordionButton {...props} />
+);
+const AccordionPanel: FC<AccordionPanelProps> = (props) => (
+  <ChakraAccordionPanel {...props} />
+);
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Accordion/Accordion.tsx:32]: this components are redundant, you can safely remove then and use the original chakra component below

File: packages/components/src/theme/ComposerTheme/Components/Accordion.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    },
+  },
+  variants: {
+    highlight: {
+      button: {
+        _expanded: {
+          background: 'background-action-active',
+        },
+      },
+    },
```
Comment (tkanzakic) [inline packages/components/src/theme/ComposerTheme/Components/Accordion.ts:16]: having a single variant doesn't make a lot of sense to me

Comment (dawinValenzuela) [reply packages/components/src/theme/ComposerTheme/Components/Accordion.ts]: removed!

File: packages/components/src/components/primitives/Accordion/Accordion.tsx (hunk 1 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/Accordion/Accordion.tsx]: done!

File: packages/components/src/components/primitives/Accordion/Accordion.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+//     spacing={4}
+//     align="stretch"
+//   >
+//     {accordionVariants.map((variant, id) => (
+//       <Template variant={variant} key={id} />
+//     ))}
+//   </VStack>
+// );
+
+// export const All = AllTemplate.bind({});
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Accordion/Accordion.stories.tsx:27]: ??

**Review by tkanzakic — APPROVED** [review]
one comment left and LGTM

File: packages/components/src/components/primitives/Accordion/Accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import {
+  AccordionProps,
+  AccordionButtonProps,
+  AccordionStaticMembers,
+} from './types';
+
+export const Accordion: FC<AccordionProps> & AccordionStaticMembers = (
+  props
+) => {
+  return <ChakraAccordion _expanded={{ bg: '#]]]' }} {...props} />;
```
Comment (guilean) [inline packages/components/src/components/primitives/Accordion/Accordion.tsx:19]: Is this `_expanded={{ bg: '#]]]' }}` needed?

File: packages/components/src/components/primitives/Accordion/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+const AccordionButton: FC<AccordionButtonProps> = ({ highlight, ...props }) => {
+  let expanded = null;
+  if (highlight) {
+    expanded = { bg: 'background-action-active' };
```
Comment (guilean) [inline packages/components/src/components/primitives/Accordion/Accordion.tsx:25]: Could we simplify this to something like this so we can remove the expanded variable?

`  {...(highlight && { bg: 'background-action-active' })}`

**Review by guilean — CHANGES_REQUESTED** [review]

File: packages/components/src/components/primitives/Accordion/Accordion.stories.tsx (hunk 3 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/Accordion/Accordion.stories.tsx]: removed !

File: packages/components/src/components/primitives/Accordion/Accordion.tsx (hunk 5 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/Accordion/Accordion.tsx]: done!

File: packages/components/src/components/primitives/Accordion/Accordion.tsx (hunk 4 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/Accordion/Accordion.tsx]: removed!

**Review by guilean — APPROVED** [review]
Add changeset and should be ready to merge

**Pull request merged**